Source/WebKit/WebProcess/Plugins/PluginView.cpp

```diff
@@ -534,7 +532,28 @@ void PluginView::paint(GraphicsContext& context, const IntRect& dirtyRect, Widge
         return;
     }
 
-    m_plugin->paint(context, dirtyRect);
+    bool isSnapshotting = [&]() {
+        auto* frameView = frame()->view();
+        if (!frameView)
+            return false;
+
+        return frameView->paintBehavior().contains(PaintBehavior::FlattenCompositingLayers);
+    }();
+
+    if (!isSnapshotting && m_plugin->layerHostingStrategy() == PluginLayerHostingStrategy::GraphicsLayer)
```

Comment:
> ... and the plugin uses PluginLayerHostingStrategy::GraphicsLayer, which is only
true for the UnifiedPDFPlugin

Would an `m_plugin->isUnifiedPDFPlugin()` be appropriate here? Or we want to do this early return whenever a plugin has that layer hosting strategy (not familiar with what this means, so this could be an unsound comment...)

Reply:
`isUnifiedPDFPlugin` would be also fine, and equivalent. But maybe some day we'll have a 3rd plugin type that uses GraphicsLayers.